Engine/AlgorithmManager.cs

```diff
@@ -1041,8 +1041,18 @@ private static void ProcessDelistedSymbols(IAlgorithm algorithm, List<Delisting>
                 var delistingTime = delistings[i].Time;
                 var nextMarketOpen = security.Exchange.Hours.GetNextMarketOpen(delistingTime, false);
                 var nextMarketClose = security.Exchange.Hours.GetNextMarketClose(nextMarketOpen, false);
+                var futureOrFutureOption = security.Type == SecurityType.Future ||
+                    security.Type == SecurityType.FutureOption;
 
-                if (security.LocalTime < nextMarketClose)
+                // Legacy behavior, we want to liquidate at the next market open for any
+                // non-futures/FOPs assets with an expiry
+                if (!futureOrFutureOption && security.LocalTime < nextMarketClose)
```

Comment:
Not fixing the original issue here, if `delistingTime` is at market open the `GetNextMarketOpen` will return the next day, missing something like:
```
                var nextMarketOpen = delistingTime;
                if (!security.Exchange.Hours.IsOpen(nextMarketOpen, false))
                {
                    nextMarketOpen = security.Exchange.Hours.GetNextMarketOpen(delistingTime, false);
                }
```
Suggest we extract this logic into a method we can unit test, or maybe it's this entire method 👍. So we can add some unit test cases for the different cases that could happen